ports/esp32/network_lan.c

```diff
@@ -97,14 +102,17 @@ STATIC mp_obj_t get_lan(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_ar
         return MP_OBJ_FROM_PTR(&lan_obj);
     }
 
-    enum { ARG_id, ARG_mdc, ARG_mdio, ARG_power, ARG_phy_addr, ARG_phy_type };
+    enum { ARG_id, ARG_mdc, ARG_mdio, ARG_power, ARG_phy_addr, ARG_phy_type, ARG_spi_id, ARG_cs, ARG_int };
     static const mp_arg_t allowed_args[] = {
         { MP_QSTR_id,           MP_ARG_OBJ, {.u_obj = mp_const_none} },
-        { MP_QSTR_mdc,          MP_ARG_KW_ONLY | MP_ARG_REQUIRED | MP_ARG_OBJ },
-        { MP_QSTR_mdio,         MP_ARG_KW_ONLY | MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_mdc,          MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+        { MP_QSTR_mdio,         MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
```

Comment:
Now that these are not required args, what happens when you don't specify them but they are needed (in the original case of using a non-SPI PHY)?  Does it give a sensible error message?

Reply:
Good point, it wouldn't. I added a manual check for this in a !IS_SPI_PHY path.

Reply:
I adjusted the code flow a bit by moving the ESP32 embedded ethernet MAC initialization at the bottom, I think it makes more sense now. (And from the look of it, it will make porting to ESP-IDF 5 simpler.)

Reply:
What is the relation to PRs  #10220 and #10329 ? I see that it is partially overlapping. The referenced PR's allow in addition to set the ref_clk_pin and ref_clk_mode.

Reply:
@robert-hh It's just moving code around. It doesn't conceptually clash with #10329 at all (but it will likely require a small manual conflict fix).